### PR TITLE
Chrome 130 supports WebAssemblyCompileOptions, which enable JS String Builtins

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

Chrome 130 supports [WebAssembly JS String builtins](https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md). Specifically, This PR adds data points for the `compileOptions` parameter used to enable these built-ins, which is added to:

- [`WebAssembly.compile()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/compile_static)
- [`WebAssembly.compileStreaming()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/compileStreaming_static)
- [`WebAssembly.instantiate()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/instantiate_static)
- [`WebAssembly.instantiateStreaming()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/instantiateStreaming_static)
- [`WebAssembly.validate()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/validate_static)
- The [`WebAssembly.Module()`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/Module/Module) constructor

#### Test results and supporting details

See https://chromestatus.com/feature/6695587390423040 for the data source. I have also tested each of the above cases and found them to work by default in Chrome 130/131.

### CODE REVIEW

File: webassembly/api.json [hunk 1]
```diff
@@ -90,6 +90,50 @@
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "compile_options": {
+          "__compat": {
+            "description": "`compileOptions` parameter",
+            "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#using-builtins",
```
Comment (caugner) [inline webassembly/api.json:97]: How about:

```suggestion
            "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#:~:text=dictionary-,webassemblycompileoptions",
```

File: webassembly/api.json [hunk 2]
```diff
@@ -139,6 +183,50 @@
...
```
Comment (caugner) [inline webassembly/api.json:190]: Same:

```suggestion
            "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#:~:text=dictionary-,webassemblycompileoptions",
```

File: webassembly/api.json [hunk 3]
```diff
@@ -186,6 +274,50 @@
...
```
Comment (caugner) [inline webassembly/api.json:281]:
```suggestion
            "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#:~:text=dictionary-,webassemblycompileoptions",
```

File: webassembly/api.json [hunk 4]
```diff
@@ -282,6 +458,50 @@
...
```
Comment (caugner) [inline webassembly/api.json:465]:
```suggestion
            "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#:~:text=dictionary-,webassemblycompileoptions",
```

File: webassembly/api/Module.json [hunk 5]
```diff
@@ -91,6 +91,50 @@
               "standard_track": true,
               "deprecated": false
             }
+          },
+          "compile_options": {
+            "__compat": {
+              "description": "`compileOptions` parameter",
+              "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#using-builtins",
```
Comment (caugner) [inline webassembly/api/Module.json:98]:
```suggestion
              "spec_url": "https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#:~:text=dictionary-,webassemblycompileoptions",
```

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM overall, just the spec_url doesn't seem specific enough.

PS: It is what we currently do, but I really don't like this data duplication (6 feature for what is likely covered by the same implementation) and would prefer a way to inherit support data from another feature.

File: webassembly/api.json [hunk 6]
```diff
@@ -90,6 +90,50 @@
...
+            "tags": [
+              "web-features:wasm"
```
Comment (Elchi3) [inline webassembly/api.json:99]: I created a web-feature for this recently. https://github.com/web-platform-dx/web-features/blob/main/features/wasm-string-builtins.yml

```suggestion
              "web-features:wasm-string-builtins"
```

File: webassembly/api.json [hunk 7]
```diff
@@ -139,6 +183,50 @@
...
```
Comment (Elchi3) [inline webassembly/api.json:192]:
```suggestion
              "web-features:wasm-string-builtins"
```

File: webassembly/api.json [hunk 8]
```diff
@@ -186,6 +274,50 @@
...
```
Comment (Elchi3) [inline webassembly/api.json:283]:
```suggestion
              "web-features:wasm-string-builtins"
```

File: webassembly/api.json [hunk 9]
```diff
@@ -235,6 +367,50 @@
...
```
Comment (Elchi3) [inline webassembly/api.json:376]:
```suggestion
              "web-features:wasm-string-builtins"
```

File: webassembly/api.json [hunk 10]
```diff
@@ -282,6 +458,50 @@
...
```
Comment (Elchi3) [inline webassembly/api.json:467]:
```suggestion
              "web-features:wasm-string-builtins"
```

File: webassembly/api/Module.json [hunk 11]
```diff
@@ -91,6 +91,50 @@
...
+              "tags": [
+                "web-features:wasm"
```
Comment (Elchi3) [inline webassembly/api/Module.json:100]:
```suggestion
                "web-features:wasm-string-builtins"
```

**Comment by chrisdavidmills** [comment]
@Elchi3 @caugner cheers gents; updates applied.

**Review by caugner — APPROVED** [review]
LGTM, thanks!

**Pull request merged**